lib/practice_exercises.rb

```diff
@@ -1,13 +1,60 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) 
+# Space Complexity: O(1) constant
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+
+  string = string.downcase
+  string = string.gsub(/[\s,:]/ ,"")
+    i = 0
+    j = string.length - 1
+    while i < j
+      if string[i] == string[j]
+        i += 1
+        j -= 1
+      else
+        return false
+      end
+  end
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+
+
+# Time Complexity: O(n^2)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍 

You're right that the code could use a refactor, but it does work.